src/client/client.ts

```diff
@@ -322,12 +331,36 @@ export class Client {
       : environment.QSTASH_URL ?? "https://qstash.upstash.io";
     const token = config?.token ?? environment.QSTASH_TOKEN;
 
+    const enableTelemetry = environment.UPSTASH_DISABLE_TELEMETRY
+      ? false
+      : config?.enableTelemetry ?? true;
+
+    // @ts-expect-error caches is not defined in the types
+    const isCloudflare = typeof caches !== "undefined" && "default" in caches;
+    const telemetryHeaders = new Headers(
+      enableTelemetry
+        ? {
+            "Upstash-Telemetry-Sdk": `upstash-qstash-js@${VERSION}`,
+            "Upstash-Telemetry-Platform": isCloudflare
+              ? "cloudflare"
+              : environment.VERCEL
+                ? "vercel"
+                : environment.AWS_REGION
+                  ? "aws"
+                  : "",
```

Comment:
did we check if these work correctly?

Reply:
Checked cloudflare and vercel, they are working